File: Lab4/pzpi-23-3-horshcharuk-nikita-lab4/Server/Elevate/Entities/TeamJoinRequest.cs
namespace Elevate.Entities;

public class TeamJoinRequest
{
    public int TeamJoinRequestID { get; set; }
    public int TeamID { get; set; }
    public int UserID { get; set; }

    public string Status { get; set; } = "Pending";

    public DateTime RequestedAt { get; set; } = DateTime.UtcNow;
    public DateTime? ProcessedAt { get; set; }
    public int? ProcessedByUserID { get; set; }

    public Team Team { get; set; } = null!;
    public User User { get; set; } = null!;
}

File: Lab5/pzpi-23-3-horshcharuk-nikita-lab5/Mobile/elevate_mobile/lib/domain/entities/team/team_join_request.dart
class TeamJoinRequest {
  final int id;
  final int teamId;
  final int userId;
  final String userFullName;
  final String status;
  final DateTime requestedAt;

  const TeamJoinRequest({
    required this.id,
    required this.teamId,
    required this.userId,
    required this.userFullName,
    required this.status,
    required this.requestedAt,
  });
}
